akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -108,8 +123,14 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
         def bufferedBytes: Int = buffer.size
 
         override def onPush(): Unit = {
-          val newData = inlet.grab()
-          buffer ++= newData
+          inlet.grab() match {
+            case newData: ByteString          ⇒ buffer ++= newData
+            case HttpEntity.Chunk(newData, _) ⇒ buffer ++= newData
+            case HttpEntity.LastChunk(_, headers) ⇒
+              val headerPairs = new VectorBuilder[(String, String)]()
+              ResponseRendering.renderHeaders(headers, headerPairs, None, log)
+              trailer = Some(ParsedHeadersFrame(streamId, endStream = true, headerPairs.result(), None))
```

Comment:
Maybe add another method for that in `ResponseRendering` that directly renders a set of headers into a `ParsedHeadersFrame`.